ios/Classes/FlutterDownloaderPlugin.m

```diff
@@ -624,79 +657,79 @@ - (void) unqueueStatusEvents {
     }
 }
 
-- (void)enqueueMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
+- (void)enqueueMethodCall:(FlutterMethodCall *)call result:(FlutterResult)result {
     NSString *urlString = call.arguments[KEY_URL];
     NSString *savedDir = call.arguments[KEY_SAVED_DIR];
     NSString *shortSavedDir = [self shortenSavedDirPath:savedDir];
     NSString *fileName = call.arguments[KEY_FILE_NAME];
     NSString *headers = call.arguments[KEY_HEADERS];
     NSNumber *showNotification = call.arguments[KEY_SHOW_NOTIFICATION];
     NSNumber *openFileFromNotification = call.arguments[KEY_OPEN_FILE_FROM_NOTIFICATION];
+    bool allowCellular = [call.arguments[KEY_ALLOW_CELLULAR] boolValue];
 
-    NSURLSessionDownloadTask *task = [self downloadTaskWithURL:[NSURL URLWithString:urlString] fileName:fileName andSavedDir:savedDir andHeaders:headers];
+    NSURLSessionDownloadTask *task = [self downloadTaskWithURL:[NSURL URLWithString:urlString] fileName:fileName andSavedDir:savedDir andHeaders:headers allowCellular:allowCellular];
```

Comment:
After integrating the changes to how the sessions are initialized (in a `if(_isolate)` condition), this always returns `nil`. The plugin is initialized twice, once in the foreground where no sessions are configured and once in the background where there are.

The `enqueue` call is made on the foreground instance though, where the sessions are missing.

I'm having trouble understanding why this is working in the example and on the `master` branch.

Reply:
Disregard this, I have found the problem, I think